src/backend/distributed/connection/placement_connection.c

```diff
@@ -700,9 +700,6 @@ FindPlacementListConnection(int flags, List *placementAccessList, const char *us
 			Assert(!placementConnection->hadDML);
 			Assert(accessType != PLACEMENT_ACCESS_DDL);
 		}
-
-		/* record the relation access mapping */
-		AssociatePlacementAccessWithRelation(placement, accessType);
```

Comment:
What about the one in `AssignPlacementListToConnection`? Can we move that somewhere else to not do it per placement?

In fact, can we eliminate `AssociatePlacementAccessWithRelation` to make sure we don't start calling it again?

Reply:
I think the current changes should meet your expectations, which I also agree btw.